### DISCUSSION TITLE
How to move from dev to production?

### DISCUSSION THREAD

**Discussion by XieKaiwen** [vikejs/vike#1616] [Discussion]
I am new to Vike, in the boilerplate code using npm init Vike, the index.js uses a if-else statement for whether the app is in dev or production. 

However my website would not be static, what are the necessary steps i need to take to move from dev to production?

**Comment by brillout** [comment]
I'm not sure I understand your question.

**Reply by XieKaiwen** [reply]
oh sorry, I shall rephrase my question:

`
const isProduction = process.env.NODE_ENV === 'production'

startServer()

async function startServer() {
  const app = express()

  app.use(compression())

  // Vite integration
  if (isProduction) {
    // In production, we need to serve our static assets ourselves.
    // (In dev, Vite's middleware serves our static assets.)
    const sirv = (await import('sirv')).default
    app.use(sirv(`${root}/dist/client`))
  } else {
    // We instantiate Vite's development server and integrate its middleware to our server.
    // ⚠️ We instantiate it only in development. (It isn't needed in production and it
    // would unnecessarily bloat our production server.)
    const vite = await import('vite')
    const viteDevMiddleware = (
      await vite.createServer({
        root,
        server: { middlewareMode: true }
      })
    ).middlewares
    app.use(viteDevMiddleware)
  }
`

for this piece of code that I found in the boilerplate code in the app scaffold, the comments said that in production we have to serve our assets ourselves. Do we have to use vite build for this? Or if the only static assets that I have are like images, and I do not want my pages pre-rendered (since they should be dynamically updated), how do I change the code to fit this? I presume we just set pre-render to false, but then my static assets would not be in dist/client right?

I saw something on the vike dev website that we should use express.static() to manually serve our assets, so is that the answer to this?
Also, just to confirm, to change isProduction, we just have to change the value of NODE_ENV in our .env file right

**Reply by brillout** [reply]
People usually figure all this on their own by exploring the examples listed at https://vike.dev/renderPage (I just added Bati 01aff83b332fd968ca098c4a891e1dc7ce5e6a89).

I understand the DX isn't newcomer friendly, and the plan is to eventually have plugins for servers such as Express.js, fastify, and Hono plugins for a more zero-boilerplate setup. Although we're currently busy with higher priorities at the moment.

So I'd suggest to try one or two more times while having a look at the examples.
